### PR TITLE
fix #28

### LINKED ISSUE #28: Too many bugs in this repo
```js
dotProp.has({foo: null}, 'foo.bar') // TypeError: Cannot read property 'bar' of null

dotProp.get({foo: 1}, 'foo.bar')  // ( in node v0.12.2 ) TypeError: Object.getOwnPropertyDescriptor called on non-object

dotProp.delete({foo: null}, 'foo.bar') // Cannot convert undefined or null to object

```

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -84,7 +87,7 @@ module.exports.has = function (obj, path) {
 	for (var i = 0; i < pathArr.length; i++) {
 		obj = obj[pathArr[i]];
 
-		if (obj === undefined) {
+		if ((obj === undefined || obj === null) && i !== pathArr.length - 1) {
```
Comment (sindresorhus) [inline index.js]: Shouldn't this too be `if (!isObj(obj)) {`?

Comment (stevemao) [inline index.js]: `obj` could be a primitive.

File: test.js [hunk 2]
```diff
@@ -164,11 +169,12 @@ test('has', t => {
 	t.is(m.has(f1, 'foo'), true);
 	t.is(m.has({foo: 1}, 'foo'), true);
 	t.is(m.has({foo: null}, 'foo'), true);
-	t.is(m.has({foo: undefined}, 'foo'), false);
+	t.is(m.has({foo: undefined}, 'foo'), true);
 	t.is(m.has({foo: {bar: true}}, 'foo.bar'), true);
 	t.is(m.has({foo: {bar: {baz: true}}}, 'foo.bar.baz'), true);
 	t.is(m.has({foo: {bar: {baz: null}}}, 'foo.bar.baz'), true);
 	t.is(m.has({foo: {bar: 'a'}}, 'foo.fake.fake2'), false);
+	t.is(m.has({foo: null}, 'foo.bar'), false);
```
Comment (sindresorhus) [inline test.js]:
```js
t.is(m.has({foo: ''}, 'foo.bar'), false);
```

Fails.

Comment (stevemao) [inline test.js]: fixed

**Pull request merged**